src/Psalm/Internal/Analyzer/Statements/Block/LoopAnalyzer.php

```diff
@@ -45,7 +45,7 @@ public static function analyze(
         Context &$inner_context = null,
         bool $is_do = false,
         bool $always_enters_loop = false
-    ) {
+    ): ?bool {
```

Comment:
I reverted that one in a previous PR because of a CI failure but it was actually unrelated